notebooks/filechecker/argo-file-checker.ipynb

```diff
@@ -0,0 +1,385 @@
+    "            URL where the service is exposed. For example:\n",
+    "            `https://livkrakentst.clusters.bodc.me/ewetchy/amrit/argo-toolbox/api/file-checker`\n",
+    "        - Only one of the URLs should be uncommented based on your environment.\n",
+    "    - Sets the default DAC for validation.\n",
+    "    - Checks the connectivity to the API.\n",
+    "5. Checking files\n",
+    "    - <p>If you need to check few selected files , select them by clicking on the <b>Upload</b> button.</p>\n",
+    "6. Optionally run the checker on all files of a deployment.\n",
+    "    - <p>If you want to check a whole deployment, select the path of the files and select <b> Run Deployment Check </b>.</p>\n",
+    "7. Results are saved to a csv file.Result of first few are shown on the console.\n",
+    "    - The name of the output file is part of configuration"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4ddcb01a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# ===============================\n",
+    "# 1️⃣ Importing necessary packages\n",
+    "# ===============================\n",
+    "\n",
+    "import logging\n",
+    "import tempfile\n",
+    "import time\n",
+    "from pathlib import Path\n",
+    "from typing import Any\n",
+    "\n",
+    "import ipywidgets as widgets\n",
+    "import pandas as pd\n",
+    "import requests\n",
+    "from IPython.display import display as ipy_display\n",
+    "from ipywidgets import FileUpload, Label, Output, VBox\n",
+    "\n",
+    "logging.basicConfig(level=logging.INFO, format=\"%(levelname)s: %(message)s\")\n",
+    "logger = logging.getLogger(__name__)\n",
+    "for handler in logger.handlers[:]:\n",
+    "    logger.removeHandler(handler)\n",
+    "logger.setLevel(logging.INFO)\n",
+    "\n",
+    "\n",
+    "# ===============================\n",
+    "# 2️⃣ Configuration\n",
+    "# ===============================\n",
+    "# Local Docker instance\n",
+    "API_BASE_URL=\"http://localhost:8000\"\n",
+    "\n",
+    "# Kubernetes test instance\n",
+    "# API_BASE_URL= \"https://livkrakentst.clusters.bodc.me/ewetchy/amrit/argo-toolbox/api/file-checker\"\n",
+    "\n",
+    "# Default DAC for validation\n",
+    "DEFAULT_DAC = \"bodc\"\n",
+    "\n",
+    "# Mount location of all the deployment files\n",
+    "# if you are running the API locally via Docker, ensure this path matches the volume mount in your Docker setup\n",
+    "# for example docker  run --rm --name argo-file-checker2 -p 8000:8000 argo-file-checker\n",
+    "\n",
+    "# Result file for full deployment checks\n",
+    "deployments_files_check_result_file=\"results/deployment_files_check_result.csv\"\n",
+    "\n",
+    "# Endpoints\n",
+    "CHECK_FILE_ENDPOINT = \"/check-files\"\n",
+    "FULL_DEPLOYMENT_CHECK_ENDPOINT = \"/check-deployment\"\n",
+    "HEALTH_ENDPOINT = \"/\"\n",
```

Comment:
To be pedantic, you're already combining these with a `/` in f-strings below, so you don't need a leading slash for these.